mesonbuild/modules/pkgconfig.py

```diff
@@ -499,23 +549,37 @@ def generate_uninstalled_cflags(libs):
             if cflags and not dataonly:
                 ofile.write('Cflags: {}\n'.format(' '.join(cflags)))
 
-    @FeatureNewKwargs('pkgconfig.generate', '0.59.0', ['unescaped_variables', 'unescaped_uninstalled_variables'])
-    @FeatureNewKwargs('pkgconfig.generate', '0.54.0', ['uninstalled_variables'])
-    @FeatureNewKwargs('pkgconfig.generate', '0.42.0', ['extra_cflags'])
-    @FeatureNewKwargs('pkgconfig.generate', '0.41.0', ['variables'])
-    @FeatureNewKwargs('pkgconfig.generate', '0.54.0', ['dataonly'])
-    @permittedKwargs({'libraries', 'version', 'name', 'description', 'filebase',
-                      'subdirs', 'requires', 'requires_private', 'libraries_private',
-                      'install_dir', 'extra_cflags', 'variables', 'url', 'd_module_versions',
-                      'dataonly', 'conflicts', 'uninstalled_variables',
-                      'unescaped_variables', 'unescaped_uninstalled_variables'})
     @typed_pos_args('pkgconfig.generate', optargs=[(build.SharedLibrary, build.StaticLibrary)])
-    def generate(self, state: ModuleState, args: T.Tuple[T.Optional[T.Union[build.SharedLibrary, build.StaticLibrary]]], kwargs):
+    @typed_kwargs(
+        'pkgconfig.generate',
+        D_MODULE_VERSIONS_KW.evolve(since='0.43.0'),
+        INSTALL_DIR_KW,
+        KwargInfo('conflicts', ContainerTypeInfo(list, str), default=[], listify=True),
+        KwargInfo('dataonly', bool, default=False, since='0.54.0'),
+        KwargInfo('description', str, default=''),
```

Comment:
Currently, if someone explicitly passes `description: ''` it will override the description from the default. After making it default to an empty string and checking `bool(kwargs['description'])` that no longer works, we should probably instead check if it is not the sentinel value None.